### DISCUSSION TITLE
What is the best way to change the default translations without them being overridden during an update

### DISCUSSION THREAD

**Discussion by forxer** [Laravel-Lang/common#92] [Q&A] [answered]
For example in the file `/lang/fr/validation.php `there is the attribute:

`'date_of_birth' => 'date de naissance',`

We want to transform it to:

`'date_of_birth' => 'anniversaire',`

But when we update `php artisan lang:update` obviously our change is overwritten.

What is the best way to modify these default translation strings without them being overwritten after an update?

**Comment by andrey-helldar** [comment]
You need to disable the use of the `laravel-lang/attributes` package. You can do this in three ways.

1. Use the [birthday](https://github.com/Laravel-Lang/attributes/blob/main/locales/fr/php.json#L7) key instead.

2. If you are using `laravel-lang/common`:
```bash
composer remove laravel-lang/common
composer require laravel-lang/lang laravel-lang/http-statuses
```

3. If you are not using `laravel-lang/common`:
```bash
composer remove laravel-lang/attributes
```

**Comment by forxer** [comment]
Ok I used a bad example: attributes. Let's take another somewhat silly example, still in the /lang/fr/validation.php file, the default value is:

'accepted' => 'Le champ :attribute doit être accepté.',

I would like to change it for example to:

'accepted' => 'Vous devez accepter le champ :attribute.',

The question is really about the title of this thread which I will rephrase as:
What is the best way to change the default translations without them being overridden during an update.

**Reply by andrey-helldar** [reply, marked as answer]
Currently, there are two options for preserving the translation entered by the developer:

1. Write your own localization publisher that has this functionality;
2. Refuse to use the Laravel Lang project.

Our publisher used to have the ability to ignore some translation keys, but we decided to abandon it due to low demand and the need to support this functionality.

**Reply by forxer** [reply]
This was not a review, I was just wondering if there was a possibility to do this.

It is therefore not possible. I expected it, but in doubt it was better to ask the question. :)
